test/tests/kernels/ad_mat_coupled_annihilation/tests

```diff
@@ -0,0 +1,10 @@
+[Tests]
+  [ad_mat_coupled_annihilation]
+    type = 'Exodiff'
+    input = 'ad_mat_coupled_annihilation_test.i'
+    exodiff = 'ad_mat_coupled_annihilation_test_out.e'
+    issues = '#35'
+    design = '/ADMatCoupledDefectAnnihilation.md'
+    requirement = 'The system shall compute the annihilation contribution in the material.'
```

Comment:
The annihilation contribution of what? Would like this to be a bit more descriptive for auditors and users who might look at this spec.